lib/MusicBrainz/Server/Entity/Artist.pm

```diff
@@ -25,6 +25,17 @@ has 'sort_name' => (
     isa => 'Str',
 );
 
+has 'aliases' => (
+    is => 'rw',
+    isa => 'ArrayRef[Any]',
```

Comment:
```suggestion
    isa => 'ArrayRef[MusicBrainz::Server::Entity::Alias]',
```

Reply:
Sorry, I forgot to leave a comment about this after uploading this commit. That's what I tried at first, but I got validation errors when performing indexed searches:

```
Attribute (aliases) does not pass the type constraint because: Validation failed for 'ArrayRef[MusicBrainz::Server::Entity::Alias]' with value [ { begin-date: undef, end-date: undef, gid: undef, id: 1, locale: "de", name: "Pjotr Iljitsch Tschaikowski", ... }, { begin-date: undef, end-date: undef, gid: undef, id: 1, locale: undef, name: "Tchaikovskij", ... }, { begin-date: undef, end-date: undef, gid: undef, id: 1, locale: undef, name: "Pyotr Il'yich Tchaikovsky", ... }, { begin-date: undef, end-date: undef, gid: undef, id: 1, locale: undef, name: "Iljitsch", ... }, { begin-date: undef, end-date: undef, gid: undef, id: 1, locale: "en", name: "Peter Tchaikovsky", ... }, { begin-date: undef, end-date: undef, gid: undef, id: 1, locale: undef, name: "Tschaikowski", ... }, ... ] at lib/MusicBrainz/Server/Data/Search.pm line 932
MusicBrainz::Server::Data::Search::external_search(?, ?, ?, ?, ?, ?, ?) called at lib/MusicBrainz/Server/Controller/Search.pm line 250
MusicBrainz::Server::Controller::Search::do_external_search(?, ?, ?, ?, ?, ?, ?, ?, ?, undef, ?, ?) called at lib/MusicBrainz/Server/Controller/Search.pm line 229
Catalyst::forward(?, ?) called at lib/MusicBrainz/Server/Controller/Search.pm line 62
Catalyst::dispatch(?) called at lib/MusicBrainz/Server.pm line 400
MusicBrainz::Server::__ANON__ at lib/MusicBrainz/Server.pm line 365
MusicBrainz::Server::with_translations(?, ?) called at lib/MusicBrainz/Server.pm line 401
Class::MOP::Method::Wrapped::__ANON__(?) called at lib/MusicBrainz/Server.pm line 415
Class::MOP::Method::Wrapped::__ANON__(?) called at lib/MusicBrainz/Server.pm line 522
```

I _think_ that it's happening because the addition of this field to `Entity::Artist` makes the aliases get validated, which then fails due to the search API not returning objects that adhere to `Entity::Alias` (e.g. no IDs).

Clearing `$data->{aliases}` in Server/Data/Search.pm's `schema_fixup` after I find the primary alias seems to prevent the validation error, but maybe that's too much of a time bomb for someone who ends up trying to use the aliases later. Then again, they probably wouldn't be able to use the aliases for much anyway due to the missing fields...

Reply:
Oh, I see. I was expecting that you'd have `$data->{aliases} = \@aliases;` after `@aliases` is declared, so that `Entity::Artist` receives the `Entity::Alias` objects when constructed. Or does that not work either?

Reply:
Huh, I thought I'd tried this, but maybe I used the array instead of an array ref -- it seems to work. I've updated the code and added a comment about how not all of the fields are present.